### ISSUE TITLE
m2e deployments with multiple targets failing

### ISSUE THREAD

**Issue by claudioweiler** [takari/aether-connector-okhttp#29]
See
- https://bugs.eclipse.org/bugs/show_bug.cgi?id=566513
- https://bugs.eclipse.org/bugs/show_bug.cgi?id=562847

Apparently the new multi thread put on io.takari.aether.connector.AetherRepositoryConnector is failing to handle 401 status on multiple tasks.

**Comment by mickaelistria** [comment]
Thanks @claudioweiler for this report and for the analysis you provided on the m2e bugs!

I don't think this is really related to m2e specifically. I imagine that just adding aether and this connector to Maven libs (to force usage of aether for upload via CLI) would cause the same issue with CLI.
The change you linked https://github.com/takari/aether-connector-okhttp/commit/32ce3f86e2b6bc292b9f07d8306bc01a0c973605#diff-aebbc2275687e86e034ea328845cca1c5b3539ca4f5025680ae1295973f5a188 is probably faulty in some case; maybe failing at propagating authentication to all threads or something like that.

Some workarounds were identified on the m2e tickets:
* Either don't use Aether at all (but aether is used by default in m2e and needs to remain in by default)
* Set `-Daether.connector.basic.threads=1`

**Comment by mickaelistria** [comment]
Looking for `aether.connector.basic.threads` in whole GitHub (for issues and PRs) brings a few more cases where this connector seems to be the root cause.

**Comment by beyerjsb** [comment]
Just leaving it here instead of creating a new issue (I can make a new issue if you want, though):
We're getting 401 - Unauthorized on classic Maven dependency downloads with aether-connector-okhttp:0.17.8 and Nexus3 - we also still have a Nexus2 that we're going to retire finally, where this problem never happens, so it's quite possible it's a timing thing. It's failing always in the same "area" of downloads, but not everytime at the same artifact, and at this place it's downloading many files in parallel (over 20), but it doesn't happen with slower network connection to the Nexus3, only with very fast connection.
I can also confirm that -Daether.connector.basic.threads=1 kind of fixes the issue.

**Comment by claudioweiler** [comment]
Hi @beyerjsb, this problem is tightly related to authentication. So, can you confirm that you can access (and navigate) your Nexus 3 instance as anonymous?

**Comment by beyerjsb** [comment]
Hi @claudioweiler, the anonymous access to Nexus is disabled. Settings.xml contains a <mirror> and accordingly configured <server> with user and password (with the use of security-settings.xml).
Also, just to make sure this is understood properly, it is not failing with a 401 for each artifact. It is starting to download fine, and at the point in the build where it starts to download more than 20 artifacts in parallel, it always fails with a 401, but at one random artifact, not always at the same, and always at a .jar, never a pom.xml. Also, this only happens with very fast connection between Maven and Nexus, and for example does not happen with a VPN between Maven and Nexus, despite it downloading the more than 20 artifacts in parallel as well.

**Comment by claudioweiler** [comment]
Hi @mosabua, can you give some directions on how to solve this so I can try to make a PR?

**Comment by cstamas** [comment]
This repo is being archived
